### PR TITLE
#3970 - Brackets are displayed incorrectly when a S-Group is added to an atom or an expanded Functional Group or Salt is added to an atom of a structure

### PR DESCRIPTION
**Pull request by ilya-asiyuk-epam**
## How the feature works? / How did you fix the issue?
(Screenshots, videos, or GIFs, if applicable)

## Check list
- [ ] unit-tests written
- [ ] e2e-tests written
- [ ] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"
- [x] reviewers are notified about the pull request

### CODE REVIEW

**Review by StarlaStarla — APPROVED** [review]

File: packages/ketcher-core/src/application/render/restruct/resgroup.ts [hunk 1]
```diff
@@ -528,10 +527,12 @@ function getBracketParamersWithCrossBondsMoreThan2OnOneAtom(
   // if bonds direction is clockwise, then negated
   const needNegated =
     Vec2.crossProduct(bondDirections[0], bondDirections[1]) > 0;
-  crossBondsValues = crossBondsValues.sort((id1, id2) => {
-    notTemplateShapeFirstAtom = Math.abs(Number(id1) - Number(id2)) === 1;
-    return notTemplateShapeFirstAtom && !needNegated ? -1 : 0;
-  });
+  notTemplateShapeFirstAtom =
+    Math.abs(Number(crossBondsValues[0]) - Number(crossBondsValues[1])) === 1;
```
Comment (StarlaStarla) [inline packages/ketcher-core/src/application/render/restruct/resgroup.ts:531]: I am just not sure if there is case that there are more than 2 elements in crossBondsValues?

Comment (ilya-asiyuk-epam) [reply packages/ketcher-core/src/application/render/restruct/resgroup.ts:531]: Sure. Above in the code base we have additional check before calling this function:
`else if (crossBondsValues.length === 2 && crossBondsPerAtom.length === 2) { `

Comment (ilya-asiyuk-epam) [reply packages/ketcher-core/src/application/render/restruct/resgroup.ts:531]: So, in any case there are 2 elements in Array

Comment (ilya-asiyuk-epam) [reply packages/ketcher-core/src/application/render/restruct/resgroup.ts:531]: Nevertheless, I added typescript tuple [number, number]

**Review by rrodionov91 — APPROVED** [review]

**Pull request merged**